RIOT-OS/RIOT
# tests: add nanocoap CLI app for client and server testing

### Contribution description
Adds a CLI test app for nanocoap client and server testing, like the gnrc_udp test. As a CLI, it allows interactive or scripted testing. It also allows us to override/augment nanocoap as needed. For example, I have coded up a fix for #9191, and need an app to test it. So, in a follow-on PR I plan to extend the server side of the test app to ignore a configurable number of retries for a confirmable request.

### Testing procedure
See [README.md](https://github.com/kb2ma/RIOT/tree/tests/nanocoap_infrastructure/tests/coap_nanocoap) for the top level commands. You can run separate instances interactively as the client and server endpoints using the tapsetup tool.

### Issues/PRs references
Depends on #10642.
